diff -up weston-10.0.0/compositor/main.c.1~ weston-10.0.0/compositor/main.c
--- weston-10.0.0/compositor/main.c.1~	2022-02-12 23:38:04.735238639 +0100
+++ weston-10.0.0/compositor/main.c	2022-02-12 23:48:43.380342258 +0100
@@ -3500,6 +3500,12 @@ wet_main(int argc, char *argv[], const s
 		fd = -1;
 	}
 
+	/* Tell common toolkits to do the right thing... */
+	setenv("QT_QPA_PLATFORM", "wayland", 1);
+	setenv("SDL_VIDEODRIVER", "wayland", 1);
+	setenv("GDK_BACKEND", "wayland", 1);
+	setenv("CLUTTER_BACKEND", "wayland", 1);
+
 	if (fd != -1) {
 		primary_client = wl_client_create(display, fd);
 		if (!primary_client) {
